packages/spatial/src/renderer/functions/useEngineCanvas.ts

```diff
@@ -36,11 +36,16 @@ export const useEngineCanvas = (ref: React.RefObject<HTMLElement> | null) => {
     const parent = ref.current as HTMLElement
 
     const canvas = document.getElementById('engine-renderer-canvas') as HTMLCanvasElement
+    const poiUI = document.getElementById('poi-ui-container') as HTMLElement
 
     const originalParent = canvas.parentElement!
     canvas.hidden = false
 
     parent.appendChild(canvas)
+    if (poiUI) {
```

Comment:
this is not the right approach. XRUI solves this issue. please do not overload our primitive utilities with feature specific logic